[Sun Yu Pei] iP

syoopie opened the pull request:
# SUPER TASK MANAGEMENT SYSTEM :D
> "Why make life hard when it can be easy" - Sun Tzu

Ever had trouble managing your tasks??? Well look no further, because this is the task manager for you! (Not to be confused with *windows task manager*. 👀 )

TaskManager is:
- hassle free!
- easy to use! (if you type really really fast)
- text based!

Steps to install:
1. Download it [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
2. Install it
3. Use it!

Features include:
- [x] Managing Tasks
- [x] Managing Deadlines
- [ ] Managing Events
- [ ] Overall just getting your life together!

Best part of all, TaskManager is **FREE**!!! `exclaimInJoy("Oh my god it's FREE????")`

On a side note, you can also use it to practice java
```java
public static void main(String args[]) {
    manageTasks();
    System.out.println("An organized person is a happy person!");
}
```

Commit:
Add Gradle support

syoopie committed:
finished level 1, added greet, echo and exit

syoopie committed:
finished add, list

syoopie committed:
changed if-else to switch

syoopie committed:
added mark as done feature

syoopie committed:
add support for todo, event and deadline

syoopie committed:
add i/o redirection testing capabilities

syoopie committed:
add some error handling

syoopie committed:
add delete

syoopie committed:
add enum for standard user inputs, but not currently in use

syoopie committed:
add saving functionality and refractor code

Code to parse text was refractored into its own function as now it is
needed for reading from a file as well.

syoopie committed:
ignore files used for testing and update .gitignore

syoopie committed:
add date and time functionality for deadline

syoopie committed:
Merge branch 'branch-Level-7' into branch-Level-8

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java

syoopie committed:
Merge branch 'branch-Level-8'

syoopie committed:
Refactor code to make it "more OOP"

syoopie committed:
Add all classes into package "duke"

syoopie committed:
add more error checking for event and deadline and Junit tests

syoopie committed:
add a file to .gitignore

syoopie committed:
add multiple javaDocs and reorder some code

syoopie committed:
change code to comply with coding standards

syoopie committed:
add find functionality

syoopie committed:
Merge branch 'branch-A-JavaDoc'

syoopie committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Ui.java

syoopie committed:
Merge branch 'branch-Level-9'

ong-hongkai commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,87 @@
+     * @param printMsg prints confirmation messages if true, does not if false
+     * @return false if previous input was to terminate ("bye"), else returns true
+     */
+    public boolean scanInputs(Scanner scan, TaskList taskList, boolean printMsg) {
```
Could the method name incorporate the usage of verbs? Perhaps inputScanner?
Shall mark subsequent method names that could have similar problems with a "method name"

ong-hongkai commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+     *
+     * @param searchString the thing that user is searching for
+     */
+    public void findTask(String searchString) {
```
method name

ong-hongkai commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,87 @@
+     */
+    public boolean scanInputs(Scanner scan, TaskList taskList, boolean printMsg) {
+        String data = scan.next();
+        switch (data.toLowerCase()) {
```
I like the way you accounted for capital letters for commands

ong-hongkai reviewed (commented):
I looked through your code and didn't manage to find many coding violations. The only potential issue was that some method names could be named better. Other than that, great job with the project!

syoopie committed:
Merge remote-tracking branch 'origin/add-gradle-support'

syoopie committed:
fix checkstyle errors and configure gradle correctly

Haozeee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+
+            // Reads from file is there is anything in it
+            Scanner fileReader = new Scanner(myFile);
+            boolean checker = true;
```
Perhaps the boolean name can be rewritten as isChecked or something similar in order to comply with the coding standards.

Haozeee commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,37 @@
+     * Constructor.
+     * @param description description of the task
+     * @throws DukeException a DukeException
+     */
```
There should be a blank line between the method description and input parameters

syoopie committed:
JavaFX Step 1

Haozeee reviewed (commented):
Overall, the code is simple and easy to follow with some minor coding violations. Also, one suggestion to improve the coding standard would be to insert more descriptive header comments for each class, so that readers can better understand what each class is supposed to do.

syoopie committed:
Implement GUI
changed implementation of parser and UI to accomodate for new GUI

syoopie committed:
Add assertions

syoopie committed:
Correct some wrong JavaDocs

syoopie committed:
Merge branch 'Branch-Level-10'

syoopie committed:
improve quality of code

syoopie committed:
add a tag function

You can now tag tasks by using the command
tag *task number* *tag*

Example
tag 1 fun
will tag task number 1 with the tag #fun

This command is still very volatile and does not have much error
checking. Program will most likely break if input is not correct.
Same goes for the delete command. Currently if an invalid index is
used as input, the program will also crash.

Might be fixed in future updates :)

syoopie committed:
Merge pull request #3 from syoopie/BCD-Tagging

add a tag function

syoopie committed:
Merge branch 'master' into Branch-A-CodeQuality

syoopie committed:
Merge pull request #2 from syoopie/Branch-A-CodeQuality

improve quality of code

syoopie committed:
Merge pull request #1 from syoopie/Branch-A-Assertions

Branch a assertions

syoopie committed:
Merge branch 'master' of https://github.com/syoopie/ip into BCD-Tagging

* 'master' of https://github.com/syoopie/ip:
  improve quality of code
  Correct some wrong JavaDocs
  Add assertions

syoopie committed:
Add error checking for delete

Delete used to not work properly and give no indication of error when
user enters an index that is invalid.

syoopie committed:
Fix the bye command

Bye command closes the application now.

syoopie committed:
add title to application and Ui picture

syoopie committed:
update README.md

added user guide

syoopie committed:
fix bugs and recommendations from reviews
